core/templates/dev/head/pages/exploration_player/HintAndSolutionModalService.js

```diff
@@ -34,15 +34,35 @@ oppia.factory('HintAndSolutionModalService', [
             function($scope, $uibModalInstance) {
               $scope.isHint = true;
               $scope.hint = HintManagerService.consumeHint();
-              $scope.gotIt = function() {
+              $scope.closeModal = function() {
                 $uibModalInstance.dismiss('cancel');
               };
             }
           ]
         });
       },
+      displayHintModalForIndex: function(index) {
+        return $uibModal.open({
+          templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+            '/pages/exploration_player/hint_and_solution_modal_directive.html'),
+          backdrop: 'static',
+          controller: [
+            '$scope', '$modalInstance',
+            function($scope, $modalInstance) {
+              $scope.isHint = true;
+              $scope.hint = (
+                index === HintManagerService.getCurrentHintIndex() ?
```

Comment:
Actually it doesn't make sense to me that the modal is doing this. Shouldn't the button appearing in the footer constitute "consuming the hint" (since that's when the hint becomes available to the learner)? The modal just gets the hint at the given index.

It might be a terminology thing, and in the new API for the HintManagerService it might be better to call it releaseNewHint() or something. Then consumeHint() basically becomes releaseNewHint() immediately followed by getHintAtIndex(). But the mobile UI shouldn't use consumeHint() at all.

Reply:
Yes, this will be done once the new API is in place.